pinot-core/src/main/java/org/apache/pinot/core/operator/query/AggregationGroupByOperator.java

```diff
@@ -58,6 +64,34 @@ public AggregationGroupByOperator(@Nonnull AggregationFunctionContext[] function
     _transformOperator = transformOperator;
     _numTotalRawDocs = numTotalRawDocs;
     _useStarTree = useStarTree;
+
+    int numColumns = groupBy.getExpressionsSize() + _functionContexts.length;
```

Comment:
Seems like this is a change in the default path, purely due to signature change for constructor of IntermediateResultBlock? Ideally, I'd want to have no changes to default path while this project is still on-going. Of course, we can take smart risks depending on how the change. I'll let you assess that for this case.

Reply:
I personally feel this much change is fine. If you/everyone feel strongly about not touching existing path, I'll separate it out

Reply:
Separated this out into AggregationGroupByOrderBy PlanNode and Operator